app/lib/system_process.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+class SystemProcess
+  attr_accessor :pid
+
+  def initialize pid
+    @pid = pid
+  end
+
+  def cpu_percent interval = 1.0
+    return 0.0 unless still_running?
+
+    first_cpu_cycles, first_process_cycles = *cpu_and_proc_cycles
+
+    sleep interval
+
+    second_cpu_cycles, second_process_cycles =*cpu_and_proc_cycles
+
+    net_process_cycles = second_process_cycles - first_process_cycles
+    net_cpu_cycles     = second_cpu_cycles     - first_cpu_cycles
+
+    if net_process_cycles.positive? && net_cpu_cycles.positive?
+      ((net_process_cycles / net_cpu_cycles.to_f) * 100.0).round 1
+    else
+      0.0
+    end
+  end
+
+  def cpu_and_proc_cycles
+    [self.class.current_cpu_cycles, current_process_cycles]
+  end
+
+  def proc_stat
+    File.read("/proc/#{pid}/stat").split if still_running?
+  end
+
+  def still_running?
+    File.exist? "/proc/#{pid}/stat"
+  end
+
+  def current_process_cycles
+    # 13 = utime / 14 = stime
+    data = proc_stat.values_at(13, 14).compact if still_running?
+
+    data.map(&:to_i).sum if data.present?
+  end
+
+  def virtual_memory
+    proc_stat[22].to_i if still_running?
+  end
+
+  def memory
+    proc_stat[23].to_i * self.class.vm_page_size if still_running?
+  end
+
+  def memory_percent
+    (memory.to_f / self.class.memory_info[:total] * 100.0).round 1 if still_running?
+  end
+
+  def command
+    @command ||= File.read "/proc/#{pid}/cmdline" if still_running?
+  end
+
+  def created_at
+    # 21 start_time after system boot
+    self.class.boot_time + (proc_stat[21].to_f / self.class.clock_ticks) if still_running?
+  end
+
+  def time_running
+    Time.now.to_i - created_at if still_running?
+  end
+
+  # System information
+  def self.boot_time
+    @@_boot_time ||= File.readlines('/proc/stat').grep(/\Abtime/).first.split.last.to_i
+  end
+
+  def self.processors
+    @@_processors ||= File.readlines('/proc/cpuinfo').grep(/\Aprocessor/).size
+  end
+
+  def self.vm_page_size
+    @@_vm_page_size ||= `getconf PAGE_SIZE`.to_i
```

Comment:
Si bien me gustan los "backticks", creo que prefiero ver algo como:
```suggestion
    @@_vm_page_size ||= %x{getconf PAGE_SIZE}.to_i
```

Me da más rápido la idea que se está ejecutando algo. Ahí si paso rápido parece un string.

Reply:
Dale, si a mi también me pasa que la mirada rápida no te indica que *no* es un string.